module/VuFind/src/VuFind/View/Helper/Root/ProxyUrl.php

```diff
@@ -66,10 +80,66 @@ public function __construct($config = null)
      */
     public function __invoke($url)
     {
-        $usePrefix = !isset($this->config->EZproxy->prefixLinks)
-            || $this->config->EZproxy->prefixLinks;
+        $useWebService = $this->config->EZproxy->prefixLinksWebServiceUrl ?? false;
+        if ($useWebService) {
+            $usePrefix = $this->checkUrl($url) ?? $this->checkConfig();
```

Comment:
On closer examination, I'm not really sure if it makes sense to use `checkConfig()` as the fallback here. It looks to me like the main purpose of the prefixLinks setting is to allow prefixing to be globally disabled. I think if somebody configures both prefixLinks and prefixLinksWebServiceUrl at the same time, that's essentially a configuration error. I suppose it doesn't necessarily hurt anything to leave this as it is, but it just feels to me like it might not be the best option. Maybe we should just fall back to true instead of falling back to checkConfig, for example...

Reply:
I disagree on this, although not strongly.  That fallback to `checkConfig()` is going to happen on any exception checking the web service, such as a connection error, which itself is likely enough from time to time (even if the services are co-hosted; things break).  So people should think about what the fallback default should be, i.e. it's better as a configured param than hard-coded, and so we may as well reuse `prefixLinks`.  Admittedly the config.ini isn't clear that it's a fallback, so I just fixed that.

Reply:
Yes, that makes sense -- I was thinking about it differently, but tweaking the comments is a good solution. :-)